### DISCUSSION TITLE
Can we share the store/atoms within multiple packages? Using lerna and want to share the global state with other packages where components can consume the same state.

### DISCUSSION THREAD

**Discussion by bhushan-patil-official** [pmndrs/jotai#1509] [Q&A]
We are working with Lerna and have multiple packages, we have some component packages in which we need to access the atoms which are in our main app. So I was thinking to share the atoms across the multiple packages. Although I am getting an error if I import the atom in the app where it should be consumed.
`export 'jotaiItems' (imported as 'jotaiItems') was not found in '@package/store'`
Even though it is present in the d.ts file in the dist
```
export declare const jotaiItems: import("jotai/esm/").PrimitiveAtom<number> & {
    init: number;
};
```

I am exporting it the library simply by
```
// in @package/store
export const jotaiItems = atom(0);
```

Another thing I was trying to do was sharing the state across the packages using redux and then using atomWithStore to update the value of the state inside the redux

I created a simple counterReducer in the `@package/store`. created a store using createStore of redux and then imported it to my app where it needs to be consumed. Everything is working fine with redux, the actions are updating the state, no unnecessary re-rendering. Now when I use atomWithStore like this

```
// App where I am consuming the states
export const atomWithStoreAtom = atomWithStore(storeWithRedux);
export const numOfItemsAtom = atom(
	(get) => get(atomWithStoreAtom).numOfItems,
	(get, set, newValue) => {
		storeWithRedux.dispatch({ type: ADD_ITEM });
	},
);
```

And then use the atom like this

```
// In component1 in consumable app
const [numItems, updateItems] = useAtom(numOfItemsAtom);
```

It updates and re-render other components as well where other states are being used from reducer and where re-render is not needed. For redux I am wrapping my app with Provider of redux and passing `storeWithRedux` to it. All the states are working fine with redux actions and are consumable properly and no re-render issues. Although when I try with atom and use store dispatch I get unnecessary re-renders. Is there something I can do?

**Comment by dai-shi** [comment]
> multiple packages

As long as the atom reference is the same, it works. Usually, it just works.

**Reply by bhushan-patil-official** [reply]
Thanks for the quick reply.

So you are saying there must be some issue in the package building process where I am exporting atoms?

**Reply by bhushan-patil-official** [reply]
I am using microbundle-crl to build the package, I was getting some error of `d.mts cannot be exported` using rollup

**Reply by dai-shi** [reply]
Perhaps.

Try:
```js
// in @package/store
export const randomNumber = Math.random();
```
and see if you get same random number on the consumer ends.

**Reply by bhushan-patil-official** [reply]
I get the same randomNumber at the consumer end

**Reply by dai-shi** [reply]
Hmm, then maybe, the context for `<Provider>` is not identical. Like you have two packages of `jotai`.

**Reply by bhushan-patil-official** [reply]
I fixed the issue of export, turns out the dist was being updated after I build the package. Thank you so much for the help.

**Comment by dai-shi** [comment]
> re-renders

Do you use console.log? Please check it with React DevTools. Or try wrapping console.log with`useEffect`. If it works as expected, it's intentional behavior.

**Comment by aulneau** [comment]
I have found that if there are multiple versions of jotai installed, that can really throw a wrench into the equality of the atoms

**Reply by bhushan-patil-official** [reply]
You mean if the version of jotai in multiple packages are different we can have problems?
